sdk/data/azcosmos/cosmos_client.go

```diff
@@ -307,7 +306,6 @@ func (c *Client) NewQueryDatabasesPager(query string, o *QueryDatabasesOptions)
 
 			return newDatabasesQueryResponse(azResponse)
 		},
-		Tracer: c.internal.Tracer(),
```

Comment:
Just so I understand what's going on here, since I'm still ramping up a bit on the Go SDK at large: 

* Previously, we tried to create two nested spans here, the default one in `azcore.Pager`, then one we created in the Fetcher.
* We wanted to be able to push custom attributes up to the one in `azcore.Pager`, hence the comment referencing #23294.
* This didn't actually create the nested span, since we disallow nested spans
* Refactoring `azcore.Pager` to support custom spans is possible, but at this point we're the only ones who need that
* Rather than doing that refactor just for us (which I agree is probably unnecessary), we're just disabling the default `azcore.Pager` spans (by passing no `Tracer`) and keeping our custom ones in the Fetcher.

All that sounds reasonable to me. I'm definitely a fan of not over-refactoring abstractions to handle cases which are, at least currently, fairly niche.

Reply:
Yep, that's exactly right!